style(app,protocol-designer,labware-library,opentrons-ai-client): update stylelint

koji opened the pull request:
<!--
Thanks for taking the time to open a Pull Request (PR)! Please make sure you've read the "Opening Pull Requests" section of our Contributing Guide:

https://github.com/Opentrons/opentrons/blob/edge/CONTRIBUTING.md#opening-pull-requests

GitHub provides robust markdown to format your PR. Links, diagrams, pictures, and videos along with text formatting make it possible to create a rich and informative PR. For more information on GitHub markdown, see:

https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax

To ensure your code is reviewed quickly and thoroughly, please fill out the sections below to the best of your ability!
-->

# Overview
update stylelint to the latest v16.25.0

this will improve our CSS code since lint can catch the following case that the version we use couldn't and we can avoid using deprecated CSS properties and invalid CSS property values.

```css
padding: -var(--spacing-2);
```


The changes I needed are in [this doc](https://opentrons.atlassian.net/wiki/spaces/RPDO/pages/5384634412/CSS+Modules+Tips).

close AUTH-2455
<!--
Describe your PR at a high level. State acceptance criteria and how this PR fits into other work. Link issues, PRs, and other relevant resources.
-->

## Test Plan and Hands on Testing

<!--
Describe your testing of the PR. Emphasize testing not reflected in the code. Attach protocols, logs, screenshots and any other assets that support your testing.
-->

## Changelog

<!--
List changes introduced by this PR considering future developers and the end user. Give careful thought and clear documentation to breaking changes.
-->

## Review requests

<!--
- What do you need from reviewers to feel confident this PR is ready to merge?
- Ask questions.
-->

## Risk assessment

<!--
- Indicate the level of attention this PR needs.
- Provide context to guide reviewers.
- Discuss trade-offs, coupling, and side effects.
- Look for the possibility, even if you think it's small, that your change may affect some other part of the system.
  - For instance, changing return tip behavior may also change the behavior of labware calibration.
- How do your unit tests and on hands on testing mitigate this PR's risks and the risk of future regressions?
- Especially in high risk PRs, explain how you know your testing is enough.
-->

mjhuff committed:
refactor(app): Wire up run setup camera preferences (#19985)

Closes EXEC-2021, EXEC-1954, EXEC-1762, and EXEC-1605

ncdiehl11 committed:
feat(step-generation): wire up tip selection in JSON step generation (#19978)

This PR updates JSON step generation for manual tip selection, which is
required for proper state tracking and error handling in the PD
application before export.

Closes AUTH-2433, Closes AUTH-2443

mjhuff committed:
feat(app): view error details associated with an error-generated image (#19986)

Closes EXEC-1993

SyntaxColoring committed:
refactor(protocol-designer): Merge DELETE_STEP and DELETE_MULTIPLE_STEPS actions (#19976)

## Overview

This simplifies some of Protocol Designer's code for deleting steps.
Goes towards EXEC-2024.

## Details

Depending on how you delete a step in Protocol Designer, it either
happens through the `DELETE_STEP` Redux action or the
`DELETE_MULTIPLE_STEPS` Redux action.

The two actions have grown a small inconsistency between them, having to
do with how they update the *selected* step. [`DELETE_MULTIPLE_STEPS`
tries to automatically select a new step to replace
it](https://github.com/Opentrons/opentrons/blob/4cd4ab98bd7fca0f8912f6d66aa31a0736ea67e1/protocol-designer/src/steplist/actions/actions.ts#L73),
whereas [`DELETE_STEP` simply clears the
selection](https://github.com/Opentrons/opentrons/blob/58619d35ebf4b59f2e69a4d7503d9ead001b3e42/protocol-designer/src/ui/steps/reducers.ts#L86).

The algorithm for automatically selecting a new step would get more
complicated with the feature of concurrent module actions, so I'm not
sure whether we'll choose to preserve that behavior, or the simpler
behavior of clearing the selection. But regardless of which behavior we
choose, it's nicer if we don't have to keep two actions in sync with
each other.

So, this removes the `DELETE_STEP` action. Instead, you can just
dispatch `DELETE_MULTIPLE_STEPS` with a 1-element list.

CaseyBatten committed:
fix(api, robot-server): Do not attempt to utilize the opentrons live stream on OT-2 (#19975)

Covers RQA-4800, RQA-4801
The OT-2 does not have the live stream service, so don't allow it to interface with the service.

mjhuff committed:
feat(app): add no images available view on the odd (#19991)

mjhuff committed:
refactor(app): Add camera usage settings to redux (#19987)

Closes EXEC-1606

mjhuff committed:
feat(app): Add info toasts on images generated during an error (#19992)

Closes EXEC-2028

mjhuff committed:
refactor(app): Block robot settings: camera settings if run exists (#19990)

koji committed:
style(app,protocol-designer,labware-library,opentrons-ai-client): update stylelint to the latest

update stylelint to the latest v16.25.0

close AUTH-2455

mjhuff committed:
app(refactor): Adjust camera feature flags (#19988)

SyntaxColoring commented on app/src/organisms/Desktop/ProtocolVisualization/Controls/controls.module.css:
```diff
@@ -1,5 +1,7 @@
 .container {
   padding: 0 var(--spacing-16) var(--spacing-8);
+  container-name: controls-container;
+  container-type: inline-size;
```
What does this do?

koji replied:
this is to use container-query for protocol visualization.
initially i started working with this branch for protocol viz but stylelint showed errors to those lines because the version we use doesn't support new CSS properties and noticed that the last update was 6 years ago.
i switched the purpose of this branch.

koji committed:
fix(app): fix unnecessary rendering issue on slideouts (#19982)

* fix(app): fix unnecessary rendering issue on slideouts

SyntaxColoring commented on components/src/atoms/StepMeter/stepmeter.module.css:
```diff
@@ -6,7 +6,7 @@
 }
 
-@media (width: 1024px) and (height: 600px) {
+@media (width = 1024px) and (height = 600px) {
```
I see that this preserves the behavior of the original code, but was this a bug in the original code? Was this supposed to be something like `(width >= 1024px) and (height >= 600px)`?

Same for all the other `@media` changes in this PR.

koji replied:
oh this change came from `format-css`. probably we should keep using the original code.

i will update them.

koji replied:
: works with `max-` and `min-` so using `=` for width and height is correct.
```css
@media (width: 1024px) and (height: 600px) {
```

SyntaxColoring commented on components/src/buttons/buttons.module.css:
```diff
@@ -1,58 +1,50 @@
-      /* from legacy --button-disabled */
-      font-weight: normal;
+    /* from legacy --button-disabled */
+    cursor: default;
```
All of the nesting changes in this file are kind of weird, aren't they? What's going on here?

Especially with the redundant lines:

```
background-color: color-mod(var(--c-bg-dark) tint(10%));
background-color: color-mod(var(--c-bg-dark) tint(5%));
```

```
color: var(--c-font-light);
color: var(--c-font-disabled);
```

SyntaxColoring commented on labware-library/src/components/LabwareDetails/styles.module.css:
```diff
@@ -81,6 +81,6 @@
 
   .well_group_title {
-    margin-left: -var(--spacing-5);
+    margin-left: var(--spacing-5);
```
Nice catch.

SyntaxColoring commented on labware-library/src/components/LabwareList/styles.module.css:
```diff
@@ -1,20 +1,21 @@
+    cursor: pointer;;;;
+    font-family: AkkoPro-Regular, 'Ropa Sans', 'Open Sans', "sans-serif;"";"";";
+    font-size: var(--fs-body-2);;;;
+    text-align: center;;;;
```
What's going on with all the semicolons here?

koji replied:
they were added by `format-css`.
i'll remove them.

SyntaxColoring commented on labware-library/src/labware-creator/styles.module.css:
```diff
@@ -9,20 +9,20 @@
-      background-color: #00f;
-    }
+    color: white;
+    color: white;
```
Same question as before. Weird nesting, and redundant property assignments.

SyntaxColoring commented on labware-library/src/styles/spacing.module.css:
```diff
@@ -42,28 +42,14 @@
   /* aspect ratios */
   --aspect: {
     position: relative;
+    position: absolute;
```
Redundant property assignments again.

In this case, can `--aspect` be deleted entirely? It looks like `--aspect-4-3`, `--aspect-1-1`, and `--aspect-item` were deleted because nothing used them, but I can't find anything that uses `--aspect`, either.

SyntaxColoring commented on labware-library/src/styles/spacing.module.css:
```diff
@@ -42,28 +42,14 @@
+  inset: 0;
 }
+
+/* padding-bottom: 100%; */
```
Leftover comment?

SyntaxColoring commented on .stylelintrc.js:
```diff
@@ -46,6 +80,8 @@ module.exports = {
           'lost-utility',
           'lost-waffle',
+          'container-name',
+          'container-type',
```
Nit: Can we add an empty line to separate these from the group above, so it's clear that the `// TODO(mc, 2018-02-09)` comment doesn't apply to these?

```suggestion

          'container-name',
          'container-type',
```

SyntaxColoring reviewed (commented):
Thank you thank you!

I noticed a handful of weird things.

jwwojak committed:
docs: Flex log files description and download (#19863)

# Overview

Having more info about Flex log files in the manual is helpful. Also,
and perhaps more important, having this in the Flex manual allows us to
get rid of another bad Help Center article ([How to download the logs on
Opentrons
Flex](https://support.opentrons.com/s/article/How-to-download-the-logs-on-Opentrons-Flex)).

For this article, we'll repurpose the intro and file download
instructions from the Flex service manual, but without all the extra
details used by the service techs.

See also RTC-824

Sandbox docs:
- Top-level section re-org:
https://sandbox.docs.opentrons.com/docs-download-log-files/flex/advanced-operation/
- Log files:
https://sandbox.docs.opentrons.com/docs-download-log-files/flex/advanced-operation/log-files/

## Test Plan and Hands on Testing

- Work through the instructions to make sure the steps are correct.
- Check to ensure images render.

## Changelog

On the surface, a simple change or PR: add steps on log file download.

I'd recommend some reorganization. This belongs with the other
"advanced" operations articles. I'd like to:

- Break up the content in `advanced-operations.md`
- Make this section a multi-doc section to now include:
- An intro: `advanced-operations/index.md` to summarize the sections.
    - Make the command line / SSH piece a separate file.
    - Make the Jupyter Notebook a separate file
    - Add this log file as a separate file

## Review requests

Consider the reorg. Is there another place to put this _without_
changing the structure?

## Risk assessment

Low, docs changes only.

---------

Co-authored-by: Ed Cormany <edward.cormany@opentrons.com>

mjhuff committed:
refactor(app): when resuming a run, do not navigate to the run log if on the camera tab (#19995)

koji committed:
remove properties from ignore list

koji commented:
@SyntaxColoring 
thank you for reviewing. i think i trust `format-css` too much. i'll check `labware-library` again.

mjhuff committed:
refactor(app): Hide livestream UI conditionally (#19994)

CaseyBatten committed:
feat(api): Ensure image capture and filters work on OT-2 (#19997)

OT-2 utilizes the `eq` equalizer filter fields instead of the `lut` Look-up-table fields of the Flex

mjhuff committed:
refactor(app): update "app settings" styling to match "robot settings" (#19999)

Closes EXEC-2030

jerader committed:
refactor(protocol-designer): keep track of allowedTips in local state (#19996)

closes AUTH-2389

neo-jesse committed:
chore(release): 8.8 release notes placeholder (#20003)

### 8.8 release notes placeholder

A little something so we can tell in the app we are upgrading on the 8.8
alpha until the actual notes are ready.

neo-jesse committed:
fix(ci, slack-notify): fixing var passing (#19960)

### Fix GitHub Actions Slack notification syntax error

Problem:
GitHub Actions workflows were failing with syntax errors when using the
simple-build-alert action:

Root Cause:
Custom GitHub Actions cannot directly access the secrets context. They
must receive secrets as input parameters.

Solution:
Updated simple-build-alert action to accept webhook_url as an input
parameter
Updated all 6 workflows that use this action to pass the webhook URL:
app-test-build-deploy.yaml
ll-test-build-deploy.yaml
opentrons-ai-production-deploy.yaml
components-test-build-deploy.yaml
pd-test-build-deploy.yaml
shared-data-test-lint-deploy.yaml

Changes:
Added webhook_url input parameter to
.github/actions/simple-build-alert/action.yml
Updated all workflow files to pass webhook_url: ${{
secrets.OT_APP_RELEASE_SLACK_NOTIFICATION_WEBHOOK_URL }} to the action

Result:
Slack notifications for build success/failure/cancellation now work
properly across all deployment workflows.

koji committed:
update form-data package (#20008)

* update form-data package

djgomez-opentrons committed:
add mac to the connectivity csv section of robo-diag (#19971)

# Overview

Adds an output of the robots WiFi and Ethernet MAC addresses to the
Robot diagnostics QC script.

Customers have requested the ability to know the MAC address of their
robot before they receive it or set it up. By logging the MAC address in
the output of the diagnostic scripts we can now look up the MAC address
of a robot based on it's serial number in our production database.

## Test Plan and Hands on Testing

Ran the test script on a Flex in the office. Checked that the MAC
addresses were contained in the output CSV.

## Changelog

Modifies the connectivity section of the flex diagnostics QC script in
hardware testing to add wifi and ethernet mac address outputs.

## Review requests



## Risk assessment

Minimal only impacts hardware testing code

Co-authored-by: Ryan Howard <ryan.howard@opentrons.com>

emilyburghardt committed:
docs: LLD and liquid classes (#19806)

<!--
Thanks for taking the time to open a Pull Request (PR)! Please make sure
you've read the "Opening Pull Requests" section of our Contributing
Guide:


https://github.com/Opentrons/opentrons/blob/edge/CONTRIBUTING.md#opening-pull-requests

GitHub provides robust markdown to format your PR. Links, diagrams,
pictures, and videos along with text formatting make it possible to
create a rich and informative PR. For more information on GitHub
markdown, see:


https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax

To ensure your code is reviewed quickly and thoroughly, please fill out
the sections below to the best of your ability!
-->

# Overview

reworking the "what's new on Flex" section > "Flex-exclusive features"
for a few reasons:

- the OT-2 and Flex comparison is no longer what's most important in the
introduction. here, I've walked back the comparison text (but still
include a small section to do so).
- this section can be useful to us to describe what's
different/exclusive to Flex without comparison. This can also be a soft
landing place for new features that aren't ready for an entire section,
but will be one day (like LLD. After 8.8, this could very well get its
own section).
-also sprinkling throughout: LLD, liquid classes, a mention of liquid
classes in QT. Trying to bridge the gap before some of these features
get more new shiny things added to them and can have their own section
(LLD mostly).

<!--
Describe your PR at a high level. State acceptance criteria and how this
PR fits into other work. Link issues, PRs, and other relevant resources.
-->

## Test Plan and Hands on Testing

<!--
Describe your testing of the PR. Emphasize testing not reflected in the
code. Attach protocols, logs, screenshots and any other assets that
support your testing.
-->

## Changelog

<!--
List changes introduced by this PR considering future developers and the
end user. Give careful thought and clear documentation to breaking
changes.
-->

## Review requests

<!--
- What do you need from reviewers to feel confident this PR is ready to
merge?
- Ask questions.
-->

## Risk assessment

<!--
- Indicate the level of attention this PR needs.
- Provide context to guide reviewers.
- Discuss trade-offs, coupling, and side effects.
- Look for the possibility, even if you think it's small, that your
change may affect some other part of the system.
- For instance, changing return tip behavior may also change the
behavior of labware calibration.
- How do your unit tests and on hands on testing mitigate this PR's
risks and the risk of future regressions?
- Especially in high risk PRs, explain how you know your testing is
enough.
-->

koji committed:
fix(app-shell): fix protocol file read function issue (#19868)

* fix(app-shell): fix protocol file read function issue

koji committed:
keep the original CSS files

koji committed:
fix lint-css errors

koji committed:
fix lint-css errors

koji committed:
Merge branch 'edge' into style_update-stylelint-to-the-latest

koji committed:
address feedback

koji committed:
Merge branch 'style_update-stylelint-to-the-latest' of https://github.com/Opentrons/opentrons into style_update-stylelint-to-the-latest

koji committed:
address feedback

koji committed:
Merge branch 'edge' into style_update-stylelint-to-the-latest

koji committed:
Update controls.module.css

koji committed:
Merge branch 'edge' into style_update-stylelint-to-the-latest

koji committed:
Merge branch 'edge' into style_update-stylelint-to-the-latest

koji committed:
resolve conflicts

koji committed:
fix css lint errors

SyntaxColoring reviewed (approved):
Thanks! It'll definitely be good to have this updated.

LGTM when those last few failures are fixed. I assume they're just merge conflicts or something.

I suspect Copilot is right about us using invalid `{ ... }` syntax. Maybe that confused stylelint and made it add those extra semicolons and do that weird nesting thing, mentioned in the comments above. We should probably fix that at some point.

koji committed:
Merge branch 'edge' into style_update-stylelint-to-the-latest

koji committed:
fix lint errors
